html/captive-portal/lib/captiveportal/Base/Controller.pm

```diff
@@ -44,6 +44,26 @@ sub showError {
     $c->detach;
 }
 
+=head2 reached_retry_limit
+
+Test if the retry limit has been reached for a session key
+If the max is undef or 0 then it will return true
+
+=cut
+
+sub reached_retry_limit {
+    my ( $self, $c, $retry_key, $max ) = @_;
+    return 1 unless $max;
```

Comment:
If you set *sms_pin_retry_limit* to 0 (as in `conf/profiles.conf.example`), the limit will always be reached and the user will never be able to validate her phone number. Is this what we want? If the limit is set to 0, I would expect the limit to be disabled.